packages/adapter-components/src/elements/ducktype/standalone_field_extractor.ts

```diff
@@ -81,25 +81,44 @@ const addFieldTypeAndInstances = async ({
     return []
   }
 
+  const getInnerType = async (elem: TypeElement): Promise<TypeElement> => (
+    isListType(elem)
+      ? elem.getInnerType()
+      : elem
+  )
+
   const elements: Element[] = []
-  const fieldType = generateType({
-    adapterName,
-    name: toNestedTypeName(typeName, fieldName),
-    entries: instancesWithValues.flatMap(inst => inst.value[fieldName]),
-    hasDynamicFields: false,
-    isSubType: true,
-    transformationConfigByType,
-    transformationDefaultConfig,
-  })
-  type.fields[fieldName].refType = createRefToElmWithValue(fieldType.type)
-  elements.push(fieldType.type, ...fieldType.nestedTypes)
+  const currentType = await type.fields[fieldName].getType()
+  if (!isObjectType(await getInnerType(currentType))) {
+    const fieldType = generateType({
+      adapterName,
+      name: toNestedTypeName(typeName, fieldName),
+      entries: instancesWithValues.flatMap(inst => inst.value[fieldName]),
+      hasDynamicFields: false,
+      isSubType: true,
+      transformationConfigByType,
+      transformationDefaultConfig,
+    })
+    type.fields[fieldName].refType = isListType(await type.fields[fieldName].getType())
+      ? createRefToElmWithValue(new ListType(createRefToElmWithValue(fieldType.type)))
```

Comment:
nit: I think we don't need the `createRefToElmWithValue` if we call the ListType ctor
```suggestion
      ? createRefToElmWithValue(new ListType(fieldType.type))
```